src/main/resources/category/java/concurrent.xml

```diff
@@ -822,44 +913,56 @@ public class Good {
+        or pmd-java:hasAnnotation('org.springframework.beans.factory.annotation.Value')
+        or (pmd-java:hasAnnotation('javax.inject.Inject') or pmd-java:hasAnnotation('jakarta.inject.Inject'))
+    )
+ ]
+(: a variable which should (potentially) be guarded, skip synchronized(this) vars :)
+//(AssignmentExpression/*[1]|VariableAccess[not(parent::SynchronizedStatement) and not(ancestor::SynchronizedStatement/ThisExpression)])[
+   (: used in a class with guarded-by literal *IS* a field (should be private final) :)
+   @Name = ancestor::ClassDeclaration//FieldDeclaration[
+        ModifierList/Annotation[pmd-java:typeIs('net.jcip.annotations.GuardedBy') or pmd-java:typeIs('javax.annotation.concurrent.GuardedBy') or pmd-java:typeIs('com.google.errorprone.annotations.concurrent.GuardedBy')]//MemberValuePair/StringLiteral[
+              substring(@Image, 2, (string-length(@Image) - 2)) = ancestor::ClassDeclaration//FieldDeclaration/VariableDeclarator/VariableId/@Name]
+           ]/VariableDeclarator/VariableId/@Name
+    (: is defined as field :)
+    /ancestor::ClassDeclaration//FieldDeclaration/VariableDeclarator/VariableId/@Name
+    (: and *IS NOT* in a synchronized block with the GuardedBy literal:)
+    and not(concat('"', ancestor::SynchronizedStatement/VariableAccess/@Name, '"') =
+       ancestor::ClassDeclaration//FieldDeclaration[pmd-java:hasAnnotation('net.jcip.annotations.GuardedBy') or pmd-java:hasAnnotation('javax.annotation.concurrent.GuardedBy') or pmd-java:hasAnnotation('com.google.errorprone.annotations.concurrent.GuardedBy')]//MemberValuePair/StringLiteral/@Image)
+]
+,
+//MethodDeclaration[
+    (: method annotated with known frameworks :)
+    not(pmd-java:hasAnnotation('org.springframework.beans.factory.annotation.Autowired')
+        or (pmd-java:hasAnnotation('javax.annotation.PostConstruct') or pmd-java:hasAnnotation('jakarta.annotation.PostConstruct'))
+        or pmd-java:hasAnnotation('org.springframework.batch.core.annotation.BeforeStep')
+        or pmd-java:hasAnnotation('org.springframework.beans.factory.annotation.Value')
+        or (pmd-java:hasAnnotation('javax.inject.Inject') or pmd-java:hasAnnotation('jakarta.inject.Inject'))
+    )
+ ]
+(: a variable which should (potentially) be guarded, skip synchronized(this) vars :)
+//(AssignmentExpression/*[1]|VariableAccess[not(parent::SynchronizedStatement) and not(ancestor::SynchronizedStatement/ThisExpression)])[
+   (: used in a class with guarded-by literal *IS NOT* a field (should be private final) :)
+   @Name = ancestor::ClassDeclaration//FieldDeclaration[
+        ModifierList/Annotation[pmd-java:typeIs('net.jcip.annotations.GuardedBy') or pmd-java:typeIs('javax.annotation.concurrent.GuardedBy') or pmd-java:typeIs('com.google.errorprone.annotations.concurrent.GuardedBy')]//MemberValuePair/StringLiteral[
+             not(substring(@Image, 2, (string-length(@Image) - 2)) = ancestor::ClassDeclaration//FieldDeclaration/VariableDeclarator/VariableId/@Name)]
```

Comment:
Kun je hier niet beter @LiteralText gebruiken? Gebruik van @Image wordt meen ik afgeraden. Dan heb je 1 character minder voor en achter. 

En misschien is gebruik van matches i.p.v. substring hier korter (maar wellicht trager)

Reply:
Replaced with @ConstValue, and Simplified